File: SIS104Menu/app/src/main/java/com/example/sis104menu/Graficos/Grafico2d.java
package com.example.sis104menu.Graficos;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.Log;
import android.view.View;

import androidx.annotation.NonNull;

public class Grafico2d  extends View {
    private static final  String TAG = "Grafico";


    public Grafico2d(Context context) {
        super(context);
    }

     public void onDraw(@NonNull Canvas canvas) {
        super.onDraw(canvas);

         canvas.drawColor(Color.BLACK);
         Paint paint = new Paint();
         paint.setColor(Color.WHITE);


         int ancho = canvas.getWidth();
         int alto = canvas.getHeight();

         Log.d(TAG, "Ancho: " + ancho + " Alto: " + alto);

         canvas.drawLine(0, alto/2, ancho, alto/2, paint);
         canvas.drawLine(ancho/2, 0, ancho/2, alto, paint);

         float limInFx = -20;
         float limSupx = 20;
         float limInfy = -20;
         float limSupy = 20;

         paint.setColor(Color.YELLOW);

         for (float x = limInFx; x <= limSupx; x+= 0.1){
             double y = fx(x);
             Log.d(TAG, "x: " + x + " y: " + y);
             double xt = (x -limInFx) / (limSupx - limInFx) * ancho;
             double yt =alto - ((y - limInfy) / (limSupy - limInfy) * alto);
             Log.d(TAG, "xt: " + xt + " yt: " + yt);
             canvas.drawCircle((float) xt, (float) yt, 3, paint);
         }

     }

    private Double fx(float x){
        return x * Math.sin(x);
    }


}

File: ProyectoKotlin/app/src/main/java/com/example/proyectokotlin/graficos/Grafico2D.kt
package com.example.proyectokotlin.graficos

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.Log
import android.view.View

class Grafico2D(context: Context) : View(context) {

    companion object {
        private const val TAG = "Grafico"
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)

        canvas.drawColor(Color.BLACK)
        val paint = Paint().apply {
            color = Color.WHITE
        }

        val ancho = canvas.width
        val alto = canvas.height

        Log.d(TAG, "Ancho: $ancho Alto: $alto")

        // Dibujar ejes X e Y
        canvas.drawLine(0f, (alto / 2).toFloat(), ancho.toFloat(), (alto / 2).toFloat(), paint)
        canvas.drawLine((ancho / 2).toFloat(), 0f, (ancho / 2).toFloat(), alto.toFloat(), paint)

        val limInFx = -20f
        val limSupx = 20f
        val limInfy = -20f
        val limSupy = 20f

        paint.color = Color.YELLOW

        var x = limInFx
        while (x <= limSupx) {
            val y = fx(x)
            Log.d(TAG, "x: $x y: $y")
            val xt = (x - limInFx) / (limSupx - limInFx) * ancho
            val yt = alto - ((y - limInfy) / (limSupy - limInfy) * alto)
            Log.d(TAG, "xt: $xt yt: $yt")
            canvas.drawCircle(xt.toFloat(), yt.toFloat(), 3f, paint)
            x += 0.1f
        }
    }

    private fun fx(x: Float): Double {
        return x * Math.sin(x.toDouble())
    }
}
